src/compiler/parser.ts

```diff
@@ -336,10 +336,10 @@ function parseDOMNode(node: Element, ctx: ParsingContext): AST | null {
 
   for (let attr of nodeAttrsNames) {
     const value = node.getAttribute(attr)!;
-    if (attr.startsWith("t-on")) {
-      if (attr === "t-on") {
-        throw new OwlError("Missing event name with t-on directive");
-      }
+    if (attr === "t-on") {
+      throw new OwlError("Missing event name with t-on directive");
+    }
+    if (attr.startsWith("t-on-")) {
```

Comment:
IMO:
- `t-on` should throw `"Unknown QWeb directive: 't-on'"`
- `t-on-` should throw `"Missing event name with t-on directive"`

```suggestion
    if (attr.startsWith("t-on-")) {
        const eventName = attr.slice(5);
        if(!eventName) {
            throw new OwlError("Missing event name with t-on directive");
        }
```

Reply:
Maybe not, but with your version `t-on-` does not throw `"Missing event name with t-on directive"`